packages/nimble-components/src/table-column/number-text/index.ts

```diff
@@ -59,18 +73,36 @@ export class TableColumnNumberText extends TableColumnTextBase {
         this.updateColumnConfig();
     }
 
+    private decimalDigitsChanged(): void {
+        // decimalDigits is only used by the 'decimal' format option. Only update the
+        // column configuration if that is the current format option.
+        if (this.format === NumberTextFormat.decimal) {
+            this.updateColumnConfig();
+        }
+    }
+
     private updateColumnConfig(): void {
-        const columnConfig: TableColumnNumberTextColumnConfig = {
-            formatter: this.createFormatter(),
-            alignment: this.determineCellContentAlignment()
-        };
-        this.columnInternals.columnConfig = columnConfig;
+        this.validator.validateDecimalDigits(this.format, this.decimalDigits);
+
+        if (this.validator.isValid()) {
+            const columnConfig: TableColumnNumberTextColumnConfig = {
+                formatter: this.createFormatter(),
+                alignment: this.determineCellContentAlignment()
+            };
+            this.columnInternals.columnConfig = columnConfig;
+        } else {
+            this.columnInternals.columnConfig = undefined;
+        }
```

Comment:
```ts
this.columnInternals.columnConfig = this.validator.isValid()
    ? {
            formatter: this.createFormatter(),
            alignment: this.determineCellContentAlignment()
       }
     : undefined;
```
Maybe? I don't feel strongly, just thought I'd offer a "less lines" possibility.

Reply:
I liked the suggestion until I realized that the typing only knows that `this.columnInternals.columnConfig` is of type `undefined | unknown` (not particularly helpful). Therefore, I think I'm going to leave this as-is so that we can get strict typing of `columnConfig` before assigning it to `this.columnInternals.columnConfig`.